test/compiler/irpasses.jl

```diff
@@ -1161,6 +1161,35 @@ let src = @eval Module() begin
     @test count(iscall((src, Core.tuple)), src.code) == 0
 end
 
+# Test that when acde_pass!() folds a PhiNode into a value with a
+# more narrow type, it adds `IR_FLAG_REFINED` to its statement. (#53390)
+let
+    code = Any[
+        # Block 1
+        GotoIfNot(false, 3),
+        # Block 2
+        nothing,
+        # Block 3
+        PhiNode(Int32[1, 2], Any[1, Argument(1)]),
+        Expr(:call, Base.inferencebarrier, SSAValue(3)),
+        ReturnNode(SSAValue(4)),
+    ]
+    ir = make_ircode(code;
+        slottypes=Any[Float64],
+        ssavaluetypes=Any[
+            Bool,
+            Nothing,
+            Union{Int64,Float64},
```

Comment:
This should also work with just `Int64` (and an Int64 argtype), since the branch in question is `Const`, but either way is fine.